admin/services/data-access/completed-check.data.service.js

```diff
@@ -40,15 +40,30 @@ completedCheckDataService.sqlCreate = async function (data) {
 
 /**
  * Updates check record with results
- * @param data
+ * @param checkCode
+ * @param completedCheck the entire JSON payload submitted by the pupil
  * @return {Promise}
  */
-completedCheckDataService.sqlAddResult = async function (data) {
-  const checkDataParam = {
-    'id': data.data.pupil.checkCode,
-    'data': data
+completedCheckDataService.sqlAddResult = async function (checkCode, completedCheck) {
+  const params = [
+    {
+      name: 'checkCode',
+      value: checkCode,
+      type: TYPES.UniqueIdentifier
+    }
+  ]
+  const sql = `SELECT id FROM ${sqlService.adminSchema}.[check] WHERE checkCode=@checkCode`
+  let result = await sqlService.query(sql, params)
+  result = R.head(result)
+  if (!result || !result.id) {
+    throw new Error(`Could not find check with checkCode:${checkCode}`)
```

Comment:
I would say that data services should NOT throw an error.  If the query result is empty `[]` `R.head()` will return undefined.

If the caller wants to throw an error based on this it can, but returning a not fund result of `undefined` is acceptable.